sympy/functions/elementary/hyperbolic.py

```diff
@@ -722,6 +719,9 @@ def taylor_term(n, x, *previous_terms):
     def _eval_rewrite_as_cosh(self, arg):
         return S.ImaginaryUnit / cosh(arg + S.ImaginaryUnit * S.Pi / 2)
 
+    def _eval_is_finite(self):
+        return None
```

Comment:
Things like these demonstrate the need for new assumptions. The range of a symbol(say over reals) can't be handled with old assumptions.
Anyways, this won't be necessary if you want to return `None`.